ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.kt

```diff
@@ -0,0 +1,300 @@
+    mapContainerViewModel.mapLocationsOfInterest.observe(this) { map.setMapLocationsOfInterest(it) }
+    mapContainerViewModel.locationLockState.observe(this) { state ->
+      onLocationLockStateChange(state, map)
+    }
+    mapContainerViewModel.cameraUpdateRequests.observe(this) { update ->
+      update.ifUnhandled { data -> onCameraUpdate(data, map) }
+    }
+    mapContainerViewModel.surveyLoadingState.observe(this) { onSurveyChange(it) }
+    homeScreenViewModel.bottomSheetState.observe(this) { state: BottomSheetState ->
+      onBottomSheetStateChange(state, map)
+    }
+    mapContainerViewModel.mbtilesFilePaths.observe(this) { map.addLocalTileOverlays(it) }
+
+    // TODO: Do this the RxJava way
+    val cameraPosition = mapContainerViewModel.getCameraPosition().value
+    cameraPosition?.let { map.moveCamera(it.target, it.zoomLevel) }
+    mapsRepository.observableMapType().observe(this) { map.mapType = it }
+  }
+
+  private fun attachCustomViews(map: MapFragment) {
+    val repositionView = LocationOfInterestRepositionView(context, map)
+    mapContainerViewModel.moveLocationOfInterestVisibility.observe(this) {
+      repositionView.visibility = it
+    }
+    binding.mapOverlay.addView(repositionView)
+
+    val polygonDrawingView = PolygonDrawingView(context, map)
+    mapContainerViewModel.getAddPolygonVisibility().observe(this) {
+      polygonDrawingView.visibility = it
+    }
+    binding.mapOverlay.addView(polygonDrawingView)
+  }
+
+  /** Opens a dialog for selecting a [MapType] for the basemap layer. */
+  private fun showMapTypeSelectorDialog() {
+    val types = mapFragment.availableMapTypes
+    NavHostFragment.findNavController(this)
+      .navigate(
+        HomeScreenFragmentDirections.actionHomeScreenFragmentToMapTypeDialogFragment(
+          types.toTypedArray()
+        )
+      )
+  }
+
+  private fun showConfirmationDialog(point: Point) {
+    AlertDialog.Builder(requireContext())
+      .setTitle(R.string.move_point_confirmation)
+      .setPositiveButton(android.R.string.ok) { _, _ -> moveToNewPosition(point) }
+      .setNegativeButton(android.R.string.cancel) { _, _ ->
+        mapContainerViewModel.setMode(MapContainerViewModel.Mode.DEFAULT)
+      }
+      .setCancelable(true)
+      .create()
+      .show()
+  }
+
+  private fun moveToNewPosition(point: Point) {
+    val locationOfInterest = mapContainerViewModel.reposLocationOfInterest
+    if (locationOfInterest.isEmpty) {
+      Timber.e("Move point failed: No locationOfInterest selected")
+      return
+    }
+    if (locationOfInterest.get().type !== LocationOfInterestType.POINT) {
+      Timber.e("Only point locations of interest can be moved")
+      return
+    }
+    val loi = locationOfInterest.get()
+    val newPointOfInterest =
+      loi.copy(
+        loi.id,
+        loi.surveyId,
+        loi.job,
+        loi.customId,
+        loi.caption,
+        loi.created,
+        loi.lastModified,
+        point
```

Comment:
Can we omit the fields that haven't changed?

Reply:
Done!